modules/explorer-api/src/main/scala/org/ergoplatform/explorer/http/api/v1/services/Mempool.scala

```diff
@@ -70,6 +70,28 @@ object Mempool {
 
     import repo._
 
+    def getUOutputsByAddress(address: Address): F[List[UOutputInfo]] = {
+      val ergoTree = addressToErgoTreeNewtype(address)
+      (for {
+        uTxInfoL <- txs
+                      .streamRelatedToErgoTree(ergoTree, 0, Int.MaxValue)
+                      .chunkN(settings.chunkSize)
+                      .through(memprops.mkTransaction)
+                      .to[List]
+      } yield uTxInfoL.flatMap(_.outputs.filter(_.ergoTree == ergoTree.value))) ||> trans.xa
```

Comment:
Why selecting outputs via transactions? You can query unconfirmed outputs from node_u_outputs directly. I believe that will be more efficient that way.